docs/build/js-client/4-advanced-ipfs.md

```diff
@@ -0,0 +1,297 @@
+
+Let's see how to use the generated code in our application. The `index.ts` file looks this way:
+
+```typescript
+import '@fluencelabs/js-client.node'; // Import the JS Client implementation. Don't forget to add this import!
+import { Fluence } from '@fluencelabs/js-client.api'; // Import the API for JS Client
+import { exists, registerIpfsClient, remove, upload_string } from './_aqua/files.js'; // Aqua compiler provides functions which can be directly imported like any normal TypeScript function.
+import { readFile } from 'node:fs/promises';
+import { strings } from '@helia/strings';
+import { CID } from 'multiformats/cid';
+import { dagJson } from '@helia/dag-json';
+import { randomTestNet } from '@fluencelabs/fluence-network-environment';
+import { makeHelia, noop, timeout } from './utils.js';
+
+async function main() {
+    await Fluence.connect(randomTestNet());
+
+    registerIpfsClient({
+        async exists(cid: string) {
+            const helia = await makeHelia();
+
+            const s = strings(helia);
+
+            const c = CID.parse(cid);
+
+            const controller = new AbortController(); // Abort controller to shut down http requests after timeout
+            const content = await timeout(s.get(c, { signal: controller.signal }), 5 /* 5 sec */, controller).catch(noop);
+
+            const result = Boolean(content);
+            await helia.stop();
+            return result;
+        },
+        async remove(cid: string) {
+            const helia = await makeHelia();
+
+            const c = CID.parse(cid);
+
+            const isPinned = await helia.pins.isPinned(c); // Pinning protects data from garbage collector in IPFS
+
+            if (isPinned) {
+                console.log('Remove pinned entry:', c.toString());
+                const pin = await helia.pins.rm(c);
+                await helia.gc();
+            }
+
+            await helia.stop();
+            return c.toString();
+        },
+        async upload(path: string) {
+            const helia = await makeHelia();
+
+            const s = strings(helia);
+
+            const buffer = await readFile(path);
+
+            const cid = await s.add(buffer.toString());
+            await helia.pins.add(cid);
+
+            await helia.stop();
+            return cid.toString();
+        },
+        async upload_string(contents: string) {
+            const helia = await makeHelia();
+
+            const s = strings(helia);
+
+            const cid = await s.add(contents);
+            await helia.pins.add(cid);
+
+            await helia.stop();
+            return cid.toString();
+        },
+        async dag_upload(path: string) {
+            const helia = await makeHelia();
+
+            const dj = dagJson(helia);
+
+            const buffer = await readFile(path);
+
+            const cid = await dj.add(buffer.toString());
+            await helia.pins.add(cid);
+
+            await helia.stop();
+            return cid.toString();
+        },
```

Comment:
Way too much repetitive TS code which is not the point of the tutorial. I think it's better to have 1-2-3 essential functions, to keep reader focused.